samcli/local/apigw/service_error_responses.py

```diff
@@ -9,6 +9,7 @@ class ServiceErrorResponses:
     _MISSING_AUTHENTICATION = {"message": "Missing Authentication Token"}
     _LAMBDA_FAILURE = {"message": "Internal server error"}
 
+    HTTP_STATUS_CODE_501 = 501
```

Comment:
Just curious, why we went with 501 error code?

Reply:
Since the functionality of invoking sam local commands with inline code is not supported on our end we use 501 as the error code.

Reply:
I was more of looking for why we choose 501 out of other 5xx error codes. But I saw that 501 is not implemented, which makes sense for this one.